cmd/gh/main.go

```diff
@@ -53,17 +54,24 @@ func main() {
 func mainRun() exitCode {
 	buildDate := build.Date
 	buildVersion := build.Version
+	hasDebug, _ := utils.IsDebugEnabled()
+
+	cmdFactory := factory.New(buildVersion)
+	stderr := cmdFactory.IOStreams.ErrOut
+
+	ctx := context.Background()
 
+	updateCtx, updateCancel := context.WithCancel(ctx)
+	defer updateCancel()
 	updateMessageChan := make(chan *update.ReleaseInfo)
 	go func() {
-		rel, _ := checkForUpdate(buildVersion)
+		rel, err := checkForUpdate(updateCtx, buildVersion)
```

Comment:
With rearranging this code so `cmdFactory` is created above the `checkForUpdate` we can now inject the api client from the factory if we wanted to.